lib/comment/view/create_comment_page.dart

```diff
@@ -0,0 +1,561 @@
+                                customImageButtonAction: () async {
+                                  if (state.status == CreateCommentStatus.imageUploadInProgress) return;
+
+                                  String imagePath = await selectImageToUpload();
+                                  if (context.mounted) context.read<CreateCommentCubit>().uploadImage(imagePath);
+                                },
+                              ),
+                            ),
+                          ),
+                          Padding(
+                            padding: const EdgeInsets.only(bottom: 8.0, left: 8.0, right: 8.0),
+                            child: IconButton(
+                              onPressed: () {
+                                if (!showPreview) {
+                                  setState(() => wasKeyboardVisible = keyboardVisibilityController.isVisible);
+                                  FocusManager.instance.primaryFocus?.unfocus();
+                                }
+
+                                setState(() => showPreview = !showPreview);
+                                if (!showPreview && wasKeyboardVisible) _bodyFocusNode.requestFocus();
+                              },
+                              icon: Icon(
+                                showPreview ? Icons.visibility_outlined : Icons.visibility,
+                                color: theme.colorScheme.onSecondary,
+                                semanticLabel: l10n.postTogglePreview,
+                              ),
+                              visualDensity: const VisualDensity(horizontal: 1.0, vertical: 1.0),
+                              style: ElevatedButton.styleFrom(backgroundColor: theme.colorScheme.secondary),
+                            ),
+                          ),
+                        ],
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  void _validateSubmission() {
+    if (isSubmitButtonDisabled) {
+      // It's disabled, check if we can enable it.
+      if (_bodyTextController.text.isNotEmpty) {
+        setState(() {
+          isSubmitButtonDisabled = false;
+        });
+      }
+    } else {
+      // It's enabled, check if we need to disable it.
+      if (_bodyTextController.text.isEmpty) {
+        setState(() {
+          isSubmitButtonDisabled = true;
+        });
+      }
+    }
+  }
+}
+
+/// Creates a widget which displays a preview of a pre-selected language, with the ability to change the selected language
+///
+/// Passing in [languageId] will set the initial state of the widget to display that given language.
+/// A callback function [onLanguageSelected] will be triggered whenever a new language is selected from the dropdown.
+class LanguageSelector extends StatefulWidget {
```

Comment:
Doesn't have to be now, but is there any way to share this with the similar create post widget?

Reply:
I think we'll make this a separate PR, so that we can apply the suggestions that @machinaeZER0 mentioned with regards to the select language input!